zephyr/lib/alloc.c

```diff
@@ -28,6 +28,7 @@
 #include <version.h>
 #include <zephyr/sys/__assert.h>
 #include <soc.h>
+#include <adsp_memory.h>
```

Comment:
What do you actually need from adsp_memory.h? You are not adding anything here to alloc.c?

Reply:
> What do you actually need from adsp_memory.h? You are not adding anything here to alloc.c?

IMR_L3_HEAP_BASE
IMR_L3_HEAP_SIZE
IMR_L3_HEAP_BASE
IMR_L3_HEAP_SIZE

taken from DeviceTree

Reply:
Ack, might be better to have the adsp_memory.h inclusion in the patch where these start to be used. But right, I get it now.

Reply:
@marcinszkudlinski And another problem. adsp_memory.h is an Intel specific file, so this would not work with e.g. nxp targets (that use Zephyr as well) and share the alloc.c code.  So I think the IMR_L3_HEAP definitions need to be made conditional on the L3_HEAP feature and perhaps pull the values directly from DT (in alloc.c).

Reply:
> adsp_memory.h is an Intel specific file, so this would not work with e.g. nxp targets (that use Zephyr as well) and share the alloc.c code. 

This "would" not work or this _did_ not work? (URL to build logs?)

I hope it's the latter otherwise we have a big IMX coverage gap in `.github/` @dbaluta